### PR TITLE
Fixed RqMultipart leading to out-of-memory

### PR DESCRIPTION
**Pull request by d-molotchko**
Fixed issue #115 
Used temp files for request storing

### CODE REVIEW

**Comment by davvd** [comment]
Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@ggajos review this one,please

**Comment by ggajos** [comment]
@davvd please fix

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 1]
```diff
@@ -214,28 +191,65 @@ private Request make(final byte[] body, final int start,
...
+    private Collection<Request> requests(final Matcher matcher,
+            final InputStream body) throws IOException {
+        final Collection<Request> requests = new LinkedList<Request>();
+        final byte[] boundary = new StringBuilder()
+            .append("\r\n--")
+            .append(matcher.group(1)).toString().getBytes();
+        int pos = 2;
+        File file = null;
+        OutputStream fos = null;
+        while (body.available() > 0) {
```
Comment (ggajos) [inline src/main/java/org/takes/rq/RqMultipart.java]: @d-molotchko this loop has terrible design, we should extract this to other object

**Comment by ggajos** [comment]
@d-molotchko we should add some test cases also

**Comment by yegor256** [comment]
@d-molotchko also, remember that we may NOT use `null` anywhere in the framework

**Comment by yegor256** [comment]
@d-molotchko I think you over-engineered the problem here. we can just save the entire request into temporary file and then work with a fileinputstream exactly the same way we worked with memory before

**Comment by d-molotchko** [comment]
@yegor256 I think no sense in it because parsed parts still in memory. I try to parse input stream into requests with fileinputstream to temporary files. I slightly refactor my code. Review please. If my modification is unnecessary, tell me. I remake it like you said.

**Comment by yegor256** [comment]
@d-molotchko `Parser` is a mutable class and we promise that all our classes are immutable in this framework

**Comment by yegor256** [comment]
@ggajos what's your take on this?

**Comment by d-molotchko** [comment]
@yegor256 Parser will be immutable. Just not all new OOP things sits in my head yet.

**Comment by yegor256** [comment]
@d-molotchko that's why I suggest to keep the design the way it was before. making it object-oriented will require 3-4 classes and will need a lot of thinking... I would suggest to leave it untouched and only change the temporary storage of data from memory to disc

**Comment by d-molotchko** [comment]
@yegor256 ok

**Comment by ggajos** [comment]
@d-molotchko I agree with @yegor256 here, apart of that this PR is too big, we should avoid so many changes in single approach.

**Comment by ggajos** [comment]
@d-molotchko any update here?

**Comment by d-molotchko** [comment]
@ggajos no, sorry

**Comment by ggajos** [comment]
@d-molotchko task was reassigned, you should close this PR

**Comment by d-molotchko** [comment]
@ggajos ok

**Comment by davvd** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by elenavolokhova** [comment]
@davvd There was no final solution provided in this PR and issue was reassigned to another developer. Quality of this one is bad.

**Comment by elenavolokhova** [comment]
@davvd @yegor256 Should QA review be paid for this ticket?

**Comment by yegor256** [comment]
@davvd quality is bad here

**Comment by davvd** [comment]
> @davvd quality is bad here

@yegor256 thank you for paying attention

**Comment by davvd** [comment]
@ggajos I added 10 mins to @elenavolokhova (for QA review) in transaction 56844025... **20 mins** added to your account (payment number `AP-24D55698G4069831U`), many thanks for your contribution! 383 hours and 38 mins spent here.... extra minutes for review comments (c=5)... +20 to your rating, your total score is [+3091](http://www.netbout.com/b/35094?open=rating)

**Pull request closed without merging**